src/test/java/BasicMAPF/Solvers/PIBT/PIBT_SolverTest.java

```diff
@@ -474,7 +474,7 @@ void emptyMapValidityStayInPlaceConstraint() {
         System.out.println(solved.readableToString());
         assertTrue(solved.solves(testInstance));
 
-        assertEquals(35, solved.sumIndividualCosts());
+        assertEquals(27, solved.sumIndividualCosts());
```

Comment:
thinks it should be 25

Reply:
This is the solution we get now:
```
Plan for agent 0
1:(3,3)->(2,3)
2:(2,3)->(1,3)
3:(1,3)->(1,2)
4:(1,2)->(0,2)
5:(0,2)->(1,2)

Plan for agent 1
1:(1,2)->(2,2)
2:(2,2)->(3,2)
3:(3,2)->(3,3)
4:(3,3)->(3,3)
5:(3,3)->(2,3)
6:(2,3)->(3,3)

Plan for agent 2
1:(5,3)->(4,3)
2:(4,3)->(3,3)
3:(3,3)->(2,3)
4:(2,3)->(1,3)
5:(1,3)->(0,3)
6:(0,3)->(0,4)
7:(0,4)->(0,5)

Plan for agent 4
1:(4,3)->(3,3)
2:(3,3)->(2,3)
3:(2,3)->(1,3)
4:(1,3)->(1,2)
5:(1,2)->(1,1)

Plan for agent 5
1:(0,4)->(0,3)
2:(0,3)->(0,2)
3:(0,2)->(0,1)
4:(0,1)->(0,0)
```

Whereas this is the solution we got before:
```
Plan for agent 0
1:(3,3)->(2,3)
2:(2,3)->(1,3)
3:(1,3)->(1,2)
4:(1,2)->(0,2)
5:(0,2)->(1,2)
6:(1,2)->(1,2)
7:(1,2)->(1,2)

Plan for agent 1
1:(1,2)->(2,2)
2:(2,2)->(3,2)
3:(3,2)->(3,3)
4:(3,3)->(3,3)
5:(3,3)->(2,3)
6:(2,3)->(3,3)
7:(3,3)->(3,3)

Plan for agent 2
1:(5,3)->(4,3)
2:(4,3)->(3,3)
3:(3,3)->(2,3)
4:(2,3)->(1,3)
5:(1,3)->(0,3)
6:(0,3)->(0,4)
7:(0,4)->(0,5)

Plan for agent 4
1:(4,3)->(3,3)
2:(3,3)->(2,3)
3:(2,3)->(1,3)
4:(1,3)->(1,2)
5:(1,2)->(1,1)
6:(1,1)->(1,1)
7:(1,1)->(1,1)

Plan for agent 5
1:(0,4)->(0,3)
2:(0,3)->(0,2)
3:(0,2)->(0,1)
4:(0,1)->(0,0)
5:(0,0)->(0,0)
6:(0,0)->(0,0)
7:(0,0)->(0,0)
```

Are the excess 2 cost that you're talking about coming from the end of the plan for agent 1? It seems that it does a back-and-forth instead of staying in place, which means we don't catch it in the trimming phase... Interesting. 

This still means the trimming is working fine, but the cost we get in this case comes down to tie-breaking, because I don't see that agent 1 has any reason why it _has_ to move and come back. Maybe it would make sense for PIBT to prefer stay moves among equivalent moves?

Reply:
you are right the trimming is working good. The back-and-forth is because this test checks what happened when there is constraint on a goal location of agents 1 (3,3 in t=5). so he moves and go back to his goal according to the heuristic.